skiko/src/jvmMain/kotlin/org/jetbrains/skiko/FrameLimiter.kt

```diff
@@ -1,28 +1,28 @@
 package org.jetbrains.skiko
 
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.delay
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
+import kotlinx.coroutines.*
+import kotlin.time.Duration.Companion.milliseconds
 import kotlin.time.ExperimentalTime
-
-private const val NanosecondsPerMillisecond = 1_000_000L
+import kotlin.time.TimeSource
 
 /**
  * Limit the duration of the frames (to avoid high CPU usage) to [frameMillis].
  * The actual delay depends on the precision of the system timer
  * (Windows has ~15ms precision by default, Linux/macOs ~2ms).
  * FrameLimiter will try to delay frames as close as possible to [frameMillis], but not greater
  */
+@OptIn(ExperimentalTime::class, ExperimentalCoroutinesApi::class)
 class FrameLimiter(
     private val coroutineScope: CoroutineScope,
     private val frameMillis: () -> Long,
-    private val nanoTime: () -> Long = System::nanoTime
+    private val dispatcherToBlockOn: CoroutineDispatcher = Dispatchers.IO.limitedParallelism(64),
```

Comment:
We can pass it as part of `coroutineScope`

Reply:
We override this dispatcher only in tests with test dispatcher that support virtual time. So I think now it's better

Reply:
We could make it internal, but it still seems to me redundant, as the usual Kotlin pattern is to pass dispatcher is by changing coroutineContext. I don't have big preference though, internal is also will be fine

Reply:
> we don't block on anything in this class

Actually, I am not completely sure, callin `currentTime()` 15x can be slow, I added `dispatcherToBlockOn` to HardwareLayer.

We have failed test, I fixed it in the last commit. Please, review it.